src/Request.ts

```diff
@@ -0,0 +1,320 @@
+	//		The prefix to apply to sort property names that are ascending
+	descendingPrefix: string;
+
+	// headers: Object
+	//		Additional headers to pass in all requests to the server. These can be overridden
+	//		by passing additional headers to calls to the store.
+	headers: {};
+
+	// rangeStartParam: String
+	//		The indicates if range limits (start and end) should be specified
+	//		in a query parameter, and what the start parameter should be.
+	//		This must be used in conjunction with the rangeCountParam
+	//		If this is not specified, the range will
+	//		included with a RQL style limit() parameter
+	rangeStartParam: string;
+
+	// rangeCountParam: String
+	//		The indicates if range limits (start and end) should be specified
+	//		in a query parameter, and what the count parameter should be.
+	//		This must be used in conjunction with the rangeStartParam
+	//		If this is not specified, the range will
+	//		included with a RQL style limit() parameter
+	rangeCountParam: string;
+
+	// selectParam: String
+	//		This will specify the query parameter to use for specifying the
+	// 		'select` properties. This will default to `select(<properties>)`
+	// 		in the query string.
+	selectParam: string;
+
+	// sortParam: String
+	//		The query parameter to used for holding sort information. If this is omitted, than
+	//		the sort information is included in a functional query token to avoid colliding
+	//		with the set of name/value pairs.
+	sortParam: string;
+
+	// target: String
+	//		The target base URL to use for all requests to the server. This string will be
+	//		prepended to the id to generate the URL (relative or absolute) for requests
+	//		sent to the server
+	target: string;
+
+	// useRangeHeaders: Boolean
+	//		The indicates if range limits (start and end) should be specified
+	//		a Range header, using items units. If this is set to true, a header
+	//		be included of the form:
+	//			Range: items=start-end
+	useRangeHeaders: boolean;
+
+	// summary:
+	//		This is a basic store for RESTful communicating with a server through JSON
+	//		formatted data. It extends dstore/Store.
+	constructor(options?: RequestStoreArgs) {
+		super(options);
+		// summary:
+		//		This is a basic store for RESTful communicating with a server through JSON
+		//		formatted data.
+		// options: dstore/JsonRest
+		//		This provides any configuration information that will be mixed into the store
+
+		//	Defaults to JSON, but other formats can be parsed by providing an alternate
+		//	parsing function. If you do want to use an alternate format, you will probably
+		//	want to use an alternate stringify function for the serialization of data as well.
+		//	Also, if you want to support parsing a larger set of JavaScript objects
+		//	outside of strict JSON parsing, you can provide dojo/_base/json.fromJson as the parse function
+		this.parse = this.parse || function (data: string) {
+			return JSON.parse(data);
+		};
+		this.headers || (this.headers = {});
```

Comment:
might want to make this defaulting consistent with the rest `this.headers = this.headers || {}`